collector/generatorreceiver/server.go

```diff
@@ -0,0 +1,126 @@
+package generatorreceiver
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net"
+	"net/http"
+
+	"github.com/lightstep/lightstep-partner-sdk/collector/generatorreceiver/internal/flags"
+	"go.opentelemetry.io/collector/component"
+	"go.uber.org/zap"
+)
+
+type httpServer struct {
+	serverType string
+	server     *http.Server
+	logger     *zap.Logger
+	config     *Config
+	fm 		   *flags.FlagManager
+}
+
+type flagHttpResponse struct {
+	Name	string `json:"name"`
+	Enabled bool `json:"enabled"`
+}
+
+func (h *httpServer) getFlags() func(w http.ResponseWriter, r *http.Request) {
+	return func(w http.ResponseWriter, r *http.Request) {
+		if r.Method != http.MethodGet {
+			w.WriteHeader(http.StatusBadRequest)
+			fmt.Fprintf(w, "bad request")
+		}
+
+		jsonFlags := make([]flagHttpResponse, 0)
+		for flagName, flagVal := range h.fm.Flags {
+			jsonFlags = append(jsonFlags, flagHttpResponse{Name: flagName, Enabled: flagVal.Enabled()})
+		}
+		resp, err := json.MarshalIndent(jsonFlags, "", "  ")
+		if err != nil {
+			w.WriteHeader(http.StatusInternalServerError)
+			fmt.Fprintf(w, "internal error: could not set attr proc: %v", err)
+			return
+		}
+		w.Header().Set("Content-Type", "application/json")
+		w.WriteHeader(http.StatusOK)
+		fmt.Fprintf(w, "%s", string(resp))
+	}
+}
+
+func (h *httpServer) setFlag() func(w http.ResponseWriter, r *http.Request) {
```

Comment:
This could be simplified a bit by turning it into a regular function:
```go
func (h *httpServer) setFlag(w http.ResponseWriter, r *http.Request) {
    f := r.URL.Query().Get("flag")
    ... etc
```
and then passing `h.setFlag` to `handler.HandleFunc` below instead of `h.setFlag()`.

Similarly with `getFlags`.